codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -107,26 +103,29 @@ class KotlinJmespathExpressionVisitor(
 
     private fun subfield(expression: FieldExpression, parentName: String): VisitedExpression {
         val member = currentShape.targetOrSelf(ctx.model).getMember(expression.name).getOrNull()
-            ?: throw CodegenException("reference to nonexistent member '${expression.name}' of shape $currentShape")
 
         val name = expression.name.toCamelCase()
-        val nameExpr = ensureNullGuard(currentShape, name)
-
-        val memberTarget = ctx.model.expectShape(member.target)
-        val unwrapExpr = when {
-            memberTarget.isEnum -> "value"
-            memberTarget.isEnumList -> "map { it.value }"
-            memberTarget.isEnumMap -> "mapValues { (_, v) -> v.value }"
-            memberTarget.isBlobShape || memberTarget.isTimestampShape ->
-                throw CodegenException("acceptor behavior for shape type ${memberTarget.type} is undefined")
-            else -> null
+        val nameExpr = addNullGuard(currentShape, name)
+        var unwrapExpr: String? = null
+
+        if (member != null) {
+            val memberTarget = ctx.model.expectShape(member.target)
+            unwrapExpr = when {
+                memberTarget.isEnum -> "value"
+                memberTarget.isEnumList -> "map { it.value }"
+                memberTarget.isEnumMap -> "mapValues { (_, v) -> v.value }"
+                memberTarget.isBlobShape || memberTarget.isTimestampShape ->
+                    throw CodegenException("acceptor behavior for shape type ${memberTarget.type} is undefined")
+                else -> null
+            }
```

Comment:
To avoid using a `var` here, you could use a null-check  on `member` like this:

```kt
val unwrapExpr = member?.let {
	val memberTarget = ctx.model.expectShape(member.target)
	when {
		memberTarget.isEnum -> "value"
		memberTarget.isEnumList -> "map { it.value }"
		memberTarget.isEnumMap -> "mapValues { (_, v) -> v.value }"
		memberTarget.isBlobShape || memberTarget.isTimestampShape ->
			throw CodegenException("acceptor behavior for shape type ${memberTarget.type} is undefined")
                else -> null
	}
}
```